### ISSUE TITLE
metadata compatibility with other implementations

### ISSUE THREAD

**Issue by novomesk** [AOMediaCodec/libavif#247]
I saved an AVIF file using libavif with EXIF and XMP metadata but I did not have success to access XMP metadata via libheif API.
I reported the file to libheif project and the developer answered that he thinks the file is not standard compliant.
I uploaded file to https://github.com/strukturag/libheif/issues/269
Can you investigate please?

**Comment by cconcolato** [comment]
I confirm the problem: 2 irefs.
<img width="756" alt="Screen Shot 2020-07-31 at 7 20 37 AM" src="https://user-images.githubusercontent.com/1830314/89044131-65d1dd00-d2fe-11ea-8d3f-d59ad0a5e285.png">

**Comment by joedrago** [comment]
[red.zip](https://github.com/AOMediaCodec/libavif/files/5008859/red.zip)

@cconcolato @novomesk How does this one look? I made a small red square that should have the same ICC, Exif, and XMP as @novomesk 's file (after I made the commit).

**Comment by novomesk** [comment]
libheif now founds 2 metadata:
First is **Exif**
Second one is **mime** with content type **application/rdf+xml**

Is it OK? Why it is mime + application/rdf+xml and not something simple like "XMP"? Is there possibility of different kind of metadata in future?

**Comment by joedrago** [comment]
Heh, I didn't write the HEIF standard, I just implemented what it said.

![Screenshot_20200801-131929~2](https://user-images.githubusercontent.com/332342/89109860-d873b300-d3f9-11ea-8c0f-f789c62ca88e.png)

I tried to make libavif's API for these things as straightforward as possible to avoid end users having to learn this level of detail from the file, but that is the correct storage location according to the standard. libheif is just showing you the guts of the file.
